RecoTracker/LSTCore/src/alpaka/PixelQuintuplet.h

```diff
@@ -0,0 +1,947 @@
+        for (unsigned int iLSModule = connectedPixelIndex[i_pLS] + globalBlockIdx[0]; iLSModule < iLSModule_max;
+             iLSModule += gridBlockExtent[0]) {
+          //these are actual module indices
+          uint16_t quintupletLowerModuleIndex = modulesInGPU.connectedPixels[iLSModule];
+          if (quintupletLowerModuleIndex >= *modulesInGPU.nLowerModules)
+            continue;
+          if (modulesInGPU.moduleType[quintupletLowerModuleIndex] == lst::TwoS)
+            continue;
+          uint16_t pixelModuleIndex = *modulesInGPU.nLowerModules;
+          if (segmentsInGPU.isDup[i_pLS])
+            continue;
+          unsigned int nOuterQuintuplets = quintupletsInGPU.nQuintuplets[quintupletLowerModuleIndex];
+
+          if (nOuterQuintuplets == 0)
+            continue;
+
+          unsigned int pixelSegmentIndex = rangesInGPU.segmentModuleIndices[pixelModuleIndex] + i_pLS;
+
+          //fetch the quintuplet
+          for (unsigned int outerQuintupletArrayIndex = globalThreadIdx[2];
+               outerQuintupletArrayIndex < nOuterQuintuplets;
+               outerQuintupletArrayIndex += gridThreadExtent[2]) {
+            unsigned int quintupletIndex =
+                rangesInGPU.quintupletModuleIndices[quintupletLowerModuleIndex] + outerQuintupletArrayIndex;
+
+            if (quintupletsInGPU.isDup[quintupletIndex])
+              continue;
+
+            float rzChiSquared, rPhiChiSquared, rPhiChiSquaredInwards, pixelRadius, quintupletRadius, centerX, centerY;
+
+            bool success = runPixelQuintupletDefaultAlgo(acc,
+                                                         modulesInGPU,
+                                                         rangesInGPU,
+                                                         mdsInGPU,
+                                                         segmentsInGPU,
+                                                         tripletsInGPU,
+                                                         quintupletsInGPU,
+                                                         pixelSegmentIndex,
+                                                         quintupletIndex,
+                                                         rzChiSquared,
+                                                         rPhiChiSquared,
+                                                         rPhiChiSquaredInwards,
+                                                         pixelRadius,
+                                                         quintupletRadius,
+                                                         centerX,
+                                                         centerY,
+                                                         static_cast<unsigned int>(i_pLS));
+            if (success) {
+              unsigned int totOccupancyPixelQuintuplets =
+                  alpaka::atomicOp<alpaka::AtomicAdd>(acc, pixelQuintupletsInGPU.totOccupancyPixelQuintuplets, 1u);
```

Comment:
Here are mote atomics whose call could be shortened, and would likely benefit from specifying the synchronization hierarchy. (I'm not going to comment more of these at this round)